src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationReaderILGen.cs

```diff
@@ -3125,14 +3125,29 @@ private void WriteElement(string source, string? arrayName, string? choiceSource
                     ilg.Ldarg(0);
                     ilg.Call(XmlSerializationReader_get_Reader);
                     ilg.Call(XmlReader_Skip);
-                    ConstructorInfo TimeSpan_ctor = typeof(TimeSpan).GetConstructor(
-                        CodeGenerator.InstanceBindingFlags,
-                        null,
-                        new Type[] { typeof(long) },
-                        null
-                        )!;
-                    ilg.Ldc(default(TimeSpan).Ticks);
-                    ilg.New(TimeSpan_ctor);
+                    if (element.Mapping.TypeDesc!.Type == typeof(TimeSpan))
+                    {
+                        ConstructorInfo TimeSpan_ctor = typeof(TimeSpan).GetConstructor(
+                            CodeGenerator.InstanceBindingFlags,
+                            null,
+                            new Type[] { typeof(long) },
+                            null
+                            )!;
+                        ilg.Ldc(default(TimeSpan).Ticks);
+                        ilg.New(TimeSpan_ctor);
+                    }
+                    else if (element.Mapping.TypeDesc!.Type == typeof(DateTimeOffset))
+                    {
+                        ConstructorInfo DateTimeOffset_ctor = typeof(DateTimeOffset).GetConstructor(
+                            CodeGenerator.InstanceBindingFlags,
+                            null,
+                            new Type[] { typeof(long), typeof(TimeSpan) },
+                            null
+                            )!;
+                        ilg.Ldc(default(DateTimeOffset).Ticks);
+                        ilg.Ldc(default(DateTimeOffset).Offset);
+                        ilg.New(DateTimeOffset_ctor);
+                    }
```

Comment:
It looks like this code is creating a default instance of DateTimeOffset using it's constructor which takesa (Ticks long, TimeSpan offset), but it's doing it using the values from a default instance. This is unnecessary and extra work, you can use ilg.InitObj instead, which is the equivalent of `default(DateTimeOffset)`. There's an implementation of what you need in `XmlSerializationWriterILGen.ILGenForCreateInstance`. You should be able to do the same for TimeSpan too.

Reply:
Good to know. Of course I was just following the TimeSpan example. I'll update the code for both.